searchapp/views.py

```diff
@@ -60,10 +61,44 @@ def student_view(request):
     return render(request, 'student_view.html', {'data': board_list})
 
 
+def edit_questions(request):
+    if request.method == 'GET':
+        board_list = Board.objects.all().values_list('board', flat=True)
+        chapters = Chapter.objects.all().values_list('chapter_name', flat=True).distinct()
```

Comment:
If you look at the way boards and grades are being queried for generating test papers, then subject, chapters and grade are referred using foreign key and not by using `distinct()`. Using `distinct()` will not allow changes to be mapped properly hence do not use it.

Reply:
In generation of test papers, the subjects and grades are queried based on the board name. Here, everything is queried as a whole, so I used distinct to not have repeating results, for example, a Science subject for both class 9 and 10, or a Science subject for ICSE and CBSE should be treated as one subject only, because we can use filters for class and board respectively, to get the desired category of questions

Also did not understand this
> Using distinct() will not allow changes to be mapped properly hence do not use it.